### ISSUE TITLE
 Header categories is not available in file 

### ISSUE THREAD

**Issue by pointia** [techdivision/import-product-url-rewrite#33]
**Describe the bug**
Since 24.1.4 it is not possible to import a file without categories.
The problem is that by default the column categories is not required but since the last changes the \TechDivision\Import\Product\UrlRewrite\Observers\ProductUrlRewriteObserver::createArtefact create an error in line 221 ($this->setValue(ColumnKeys::CATEGORIES, $this->adminRow[$sku][ColumnKeys::CATEGORIES]);)

I think this line should only be executed if categories exist and therefore it might be better to add an additional condition to the if statement like:

```
if (!$this->hasValue(ColumnKeys::CATEGORIES) && $this->adminRow[$sku][ColumnKeys::CATEGORIES]) {
    $this->setValue(ColumnKeys::CATEGORIES, $this->adminRow[$sku][ColumnKeys::CATEGORIES]);
}
```

**To Reproduce**
1. Try to import products without categories

**Expected behavior**
Execution does work and products are imported / updated

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] Ubuntu 20.04 (Docker)

**Comment by wagnert** [comment]
Hi @pointia, thanks for supporting us. We'll check this and come back to you :-)

**Comment by wagnert** [comment]
Hi @pointia, I've tried it with 3.8.47 as well as the master branch (4.0.0-alpha) and it seems to work. Can you please provide an example file so that we can reproduce the issue? Thanks :-)

**Comment by pointia** [comment]
Hi @wagnert, 
Thanks for the try. 
The following file should do the trick. 

In general just don't add a column that is named "categories". If it does not exist we get the error and that is the reason why i was suggesting the hasValue change.

[product.csv](https://github.com/techdivision/import-product-url-rewrite/files/6799586/product.csv)

**Comment by Mardl** [comment]
Fixed in last version 24.1.8 and implement in https://github.com/techdivision/import-cli-simple/releases/tag/3.8.53
